sanitize_html/lib/src/html_formatter.dart

```diff
@@ -0,0 +1,67 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'dart:convert';
+
+import 'package:html/dom.dart';
+
+final _attrEscape = HtmlEscape(HtmlEscapeMode.attribute);
+final _textEscape = HtmlEscape(HtmlEscapeMode.unknown);
+
+String formatHtmlNode(Node node) {
+  return _HtmlFormatter()._format(node);
+}
+
+class _HtmlFormatter {
+  final _sb = StringBuffer();
+
+  String _format(Node node) {
+    _writeNodes([node], 0);
+    return _sb.toString();
+  }
+
+  void _writeNodes(List<Node> nodes, int level) {
+    if (nodes == null || nodes.isEmpty) return;
+    for (Node node in nodes) {
+      if (node is Element) {
+        _writeElement(node, level);
+      } else if (node is Text) {
+        _sb.write(_textEscape.convert(node.text));
+      } else if (node is Document) {
+        _writeNodes([node.documentElement], level);
+      } else if (node is DocumentFragment) {
+        _writeNodes(node.nodes, level);
+      } else {
+        throw UnimplementedError('Unknown node: ${node.runtimeType} ($node)');
```

Comment:
Maybe `assert(false, 'unknown node type')` followed by `return;` would be better?

That way, a user will never be able to crash the app..  but we might ignore unknown node types in production.

Reply:
In fact implementors of `Node` is:
> Comment Document DocumentFragment DocumentType Element Text

See: https://pub.dev/documentation/html/latest/dom/Node-class.html

I don't think it's important to support `DocumentType` and `Comment`, but maybe it's good for completeness..

Reply:
How about `assert(false)`, instead of throwing `UnimplementedError`, because `assert` will only cause crashes in debug mode, where as `UnimplementedError` will cause crashes in production...

Reply:
ack